quilt/tools/command.py

```diff
@@ -250,21 +252,45 @@ def check(path=None, env='default'):
     # (if not, then require dry_run=True if files!=None/all)
     build("dry_run/dry_run", path=path, dry_run=True, env=env)
 
+def _clone_git_repo(url, branch, dest):
+    cmd = ['git', 'clone', '-q', '--depth=1']
+    if branch:
+        cmd += ['-b', branch]
+    cmd += [url, dest]
+    proc = subprocess.check_call(cmd)
+
+def _rmtree(path):
+    rmtree(path)
+
 def build(package, path=None, dry_run=False, env='default'):
     """
     Compile a Quilt data package, either from a build file or an existing package node.
     """
-    # we may have a path, PackageNode, or None
+    # we may have a path, git URL, PackageNode, or None
     if isinstance(path, string_types):
-        build_from_path(package, path, dry_run=dry_run, env=env)
+        # is this a git url?
+        is_git_url = GIT_URL_RE.match(path)
+        if is_git_url:
+            tmpdir=tempfile.mkdtemp()
+            url, branch = is_git_url.groups()
+            try:
+                _clone_git_repo(url, branch, tmpdir)
+                build_from_path(package, tmpdir, dry_run=dry_run, env=env)
+            except Exception as exc:
+                print("Error %s while cloning %s.", exc, path)
```

Comment:
Please actually test this. Need `%` to substitute the values. Also, `raise CommandException` instead of `print` and `raise`.

Actually, add a unit test, too. This might come up easily if e.g. `git` is not installed, or the URL is invalid.